clients/src/main/java/org/apache/kafka/clients/admin/AdminClientConfig.java

```diff
@@ -154,10 +156,16 @@
                                         CONNECTIONS_MAX_IDLE_MS_DOC)
                                 .define(RETRIES_CONFIG,
                                         Type.INT,
-                                        5,
-                                        atLeast(0),
+                                        Integer.MAX_VALUE,
```

Comment:
Just to clarify my understanding, we will now retry "forever" with a lower request timeout (30s), but the total time spent attempting to satisfy a request will be bound by the new `default.api.timeout.ms` config _or_ the explicit timeout given in the request options (if set)

Reply:
Yes, the number of retries is unbounded by default, but the time spent retrying now has a proper bound (60s).